docs/Developers/Linter_Bears_Advanced.rst

```diff
@@ -41,6 +40,26 @@ The path of the temporary configuration file can be accessed inside
         def create_arguments(filename, file, config_file):
             return "--use-config", config_file
 
+If you return ``None``, no configuration file is generated. A common case
+where to explicitly return ``None`` is when you want to expose a setting
+for the user to use his own tool-specific config. In case the user specifies
+such a config file, we can avoid generating once again with
+``generate_config`` to reduce I/O load.
+
+::
+
+    @linter(executable='...')
+    class MyBear:
+        def generate_config(filename, file, setting_a: bool):
```

Comment:
You have to annotate `generate_config` with `@staticmethod`, as you don't have `self` in the signature